src/lexer.coffee

```diff
@@ -292,9 +292,13 @@ exports.Lexer = class Lexer
 
   # Matches JavaScript interpolated directly into the source via backticks.
   jsToken: ->
-    return 0 unless @chunk.charAt(0) is '`' and match = JSTOKEN.exec @chunk
-    @token 'JS', (script = match[0])[1...-1], 0, script.length
-    script.length
+    return 0 unless @chunk.charAt(0) is '`' and
+      (match = HERE_JSTOKEN.exec(@chunk) or JSTOKEN.exec(@chunk))
+    [js] = match
+    script = if js[0..2] is '```' then js[3...-3] else js[1...-1]
```

Comment:
This behaviour makes sense to me, I'd describe it as:

> A backslash (`\`) can be used to escape a backtick (`` ` ``) or another backslash in a JS literal, and will otherwise be passed through.
> ```coffee
`hello`                                       # hello;
`\`hello\``                                   # `hello`;
`\`Escaping backticks in JS: \\\`hello\\\`\`` # `Escaping backticks in JS: \`hello\``;
`Single backslash: \ `                        # Single backslash: \ ;
`Double backslash: \\\\`                      # Double backslash: \\;
```